docs/source/dataframe-hive.rst

```diff
@@ -0,0 +1,227 @@
+
+
+Reading Parquet Data with Hive Partitioning
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In most cases, :func:`read_parquet` will process hive-partitioned
+data automatically. By default, all hive-partitioned columns will
+be interpreted as categorical columns.
+
+.. code-block:: python
+
+    >>> ddf = dd.read_parquet("output-path", columns=["year", "semester"])
+
+    >>> ddf
+    Dask DataFrame Structure:
+                            year         semester
+    npartitions=4                                  
+                category[known]  category[known]
+                            ...              ...
+                            ...              ...
+                            ...              ...
+                            ...              ...
+    Dask Name: read-parquet, 1 graph layer
+
+    >>> ddf.compute()
+    year semester
+    0  2022     fall
+    1  2022     fall
+    2  2022     fall
+    3  2022   spring
+    4  2022   spring
+    5  2022   spring
+    6  2023     fall
+    7  2023     fall
+
+
+Defining a Custom Partitioning Schema
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When utilizing ``engine='pyarrow'``, it is possible to specify a custom
+schema for the hive-partitioned columns.
+
+.. code-block:: python
+
+    >>> schema = pa.schema([("year", pa.int16()), ("semester", pa.string())])
+
+    >>> ddf2 = dd.read_parquet(
+    ...     path,
+    ...     columns=["year", "semester"],
+    ...     dataset={"partitioning": {"flavor": "hive", "schema": schema}}
+    ... )
+    Dask DataFrame Structure:
+                    year semester
+    npartitions=4                
+                int16   object
+                    ...      ...
+                    ...      ...
+                    ...      ...
+                    ...      ...
+
+
+If any of your hive-partitioned columns contain null values, you
+**must** specify the partitioning schema in this way.
+
+Although it is not required, we also recommend that you specify
+the partitioning schema if you need to partition on high-cardinality
+columns. This is because the default ``'category'`` dtype will
+track the known categories in a way that can significantly increase
+the overall memory footprint of your Dask collection.
+
+
+Best Practices
+--------------
+
+Although hive partitioning can sometimes improve read performance
+by simplifying filtering, it can also lead to degraded performance
+and errors in other cases.
+
+
+Avoid High Cardinality
+~~~~~~~~~~~~~~~~~~~~~~
+
+The most common cause of poor user experience with hive partitioning
+is high-cardinality of the partitioning column(s). For example, if 
+you try to partition on a column with millions of unique values, then
+`:func:`to_parquet`` will need to generate millions of directories. The
+management of these directories is likely to put strain on the file
+system, and the need for many small files within each directory
+is sure to compound the issue.
```

Comment:
Not sure if I underestimate our users but is it worth mentioning that most float columns are falling into this category? Does the common user know what high cardinality is? I would hate for somebody to have a bad experience because they think their data is bound do `[0, 1)` or smth like this :)